sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,355 @@
+    }
+
+    Ok(out)
+}
+
+/// Transforms a PEM object into a PEM formatted string
+fn pem_to_string(pem: &x509_parser::pem::Pem) -> Result<String> {
+    let mut out = String::new();
+
+    // A comment will be added before the PEM formatted string to identify the certificate.
+    if let Some(comment) = comment_for_pem(&pem)? {
+        writeln!(out, "# {}", comment).context(error::WritePEMString)?;
+    }
+
+    writeln!(out, "{} {}{}", PEM_HEADER, pem.label, PEM_SUFFIX).context(error::WritePEMString)?;
+    let encoded = base64::encode(&pem.contents);
+    let bytes = encoded.as_bytes();
+    for chunk in bytes.chunks(64) {
+        let chunk = String::from_utf8_lossy(chunk);
+        writeln!(out, "{}", chunk).context(error::WritePEMString)?;
+    }
+    writeln!(out, "{} {}{}", PEM_FOOTER, pem.label, PEM_SUFFIX).context(error::WritePEMString)?;
+
+    Ok(out)
+}
+
+/// Returns a string from the common name, organizational unit or organization
+/// fields in the certificate
+fn comment_for_pem(pem: &x509_parser::pem::Pem) -> Result<Option<String>> {
+    let cert = pem.parse_x509().context(error::ParseX509Certificate)?;
+    let subject = cert.tbs_certificate.subject;
+    let comment = subject
+        .iter_common_name()
+        .chain(subject.iter_organizational_unit())
+        .chain(subject.iter_organization())
+        .next();
+
+    Ok(comment
+        .and_then(|c| c.as_str().ok())
+        .and_then(|c| Some(c.to_string())))
+}
+
+async fn run() -> Result<()> {
+    let args: Args = argh::from_env();
+
+    // SimpleLogger will send errors to stderr and anything less to stdout.
+    SimpleLogger::init(args.log_level, LogConfig::default()).context(error::Logger)?;
+
+    info!("certdog started");
+    let certificate_bundles = get_certificate_bundles(&args.socket_path).await?;
+    info!("Got certificate bundles from API");
+    update_trusted_store(certificate_bundles, args.trusted_store, args.source_bundle)?;
+    info!("Updated trusted store");
+
+    Ok(())
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+#[tokio::main]
+async fn main() {
+    if let Err(e) = run().await {
+        error!("{}", e);
+        process::exit(1);
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+mod error {
+    use snafu::Snafu;
+    use std::io;
+    use std::path::PathBuf;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub(super)")]
+    pub(super) enum Error {
+        #[snafu(display("Error sending {} to {}: {}", method, uri, source))]
+        APIRequest {
+            method: String,
+            uri: String,
+            source: apiclient::Error,
+        },
+
+        #[snafu(display("Unable to decode base64 from certificate '{}': '{}'", name, source))]
```

Comment:
Nit: we always have the 'thing' in single quotes and the error at the end without quotes.

```suggestion
        #[snafu(display("Unable to decode base64 from certificate '{}': {}", name, source))]
```